### ISSUE TITLE
Accessing inventory items during a monster’s attack appears to allow it “extra” attacks.

### ISSUE THREAD

**Issue by GoogleCodeExporter** [alistairreilly/andors-trail#132]
```
What steps will reproduce the problem?
1. Attack a monster that has more than one attack per turn.
2. During the monster’s turn, after the monster’s first attack* but before 
it completes its turn, open your inventory and select an item (so that the item 
pop-up appears).
3. Close the item pop-up and exit the inventory screen.

What is the expected output? What do you see instead?
Expected result: Monster continues its turn from where it was interrupted.

Actual result: Monster appears to start its turn from the beginning (thus 
gaining “extra” attacks).

What version of the product are you using? On what operating system?
Andor’s Trail v0.6.7 on an HTC Desire running Android 2.2 (with HTC’s Sense 
UI if that makes any difference).

Please provide any additional information below.
The procedure described above also causes the combat output window to close. 
This made it more difficult for me to keep track of the monster’s attacks 
while I was testing, so I may be misinterpreting what’s actually going on. 
Sorry if this is the case.

Accessing the inventory without selecting an item appears to work as expected.

Monsters tested: Rabid Fox (east of Crossglen), Skeletal Warrior (?) (under 
abandoned house north of Fallhaven), and Fox (on the map with the prison).

Inventory items tested: Meat, Regular potion of health, and Bone.

If this is a bug, I’m guessing it should probably be of low priority due to 
its relative obscurity and non-game breaking nature (the worst it could do is 
result in the character’s death).

Obvious workaround: Don’t access inventory items when it’s not your turn. 
;-)

*I’m not fast enough to open my inventory before the result of the 
monster’s first attack is displayed.

Random comments: Thank you for making this game. I am enjoying it and looking 
forward to seeing where you decide to take it as development continues.

Also, if it’s of any use or interest to you “Insect shell” is called 
chitin.

```

Original issue reported on code.google.com by `Grok4grok@gmail.com` on 17 Jan 2011 at 2:01

**Comment by GoogleCodeExporter** [comment]
```
Using items requires AP, really you should not be able to use items during the 
monsters attack phase. If keeping the ability to use items during the monsters 
attack phase, allowing the monster to continue to attack based on the amount of 
AP you have used seems reasonable to me.

WTF is chitin? It sounds like something you'd say to someone who has offended 
you, or you are trying to offend... This is a game first, learning is not what 
I play games for ;-)
```

Original comment by `divilsp...@gmail.com` on 17 Jan 2011 at 9:03

**Comment by GoogleCodeExporter** [comment]
```
You are correct regarding using items during the monsters attack phase - the 
game doesn't allow you to actually use them (the "use" button is greyed out for 
the useable items). However you can select any item in your inventory as 
described (there appears to be no AP cost for just looking through your 
inventory).

Lol. No, I'm not trying to offend anyone regarding chitin (it's pronounced 
kite-in, so it doesn't really sound like swearing) nor am I trying to force 
anybody to learn when they're busy having fun. ;-) I mentioned it on the 
off-chance that the developer might be interested in an alternate name for 
"Insect shell".

A correction to my original post: the "Fox" on the prison map is actually a 
Wolf (NW corner of the map).
```

Original comment by `Grok4grok@gmail.com` on 17 Jan 2011 at 10:23

**Comment by GoogleCodeExporter** [comment]
```
I totally misread your report. I thought you were saying using an item caused 
the problem. Viewing the item should not cause any problems. Do you have a 
savegame that can reproduce this? I have not been able to duplicate the problem 
when viewing my inventory. Though I was able to confirm that you cannot use 
items when it is not your turn which is something I hadn't tried before.

I am also running v0.6.7
```

Original comment by `divilsp...@gmail.com` on 18 Jan 2011 at 5:40

**Comment by GoogleCodeExporter** [comment]
```
Oh dear. I hope it's not a "bug" that's unique to my phone.

Anyway, here's my most recent savegame. I could reproduce the problem using it. 
(Remember to backup your own savegames before trying it.)

I headed South for two maps and engaged a Wild Fox (whose attack cost is listed 
as 5AP, so it gets two attacks per turn).

By opening my inventory at just the right time and selecting (tapping) an item 
(a Ruby gem this time) I could get the fox to attack me Four times.

I exited the game (without saving), reloaded and repeated the process, I got 
the same result.

I then allowed the fox to attack me uninterrupted. It got the expected two 
attacks before my next turn started.

On my last test, I decided to see how many times I could repeat the process 
within the one (fox) turn. I managed to get the fox enough "free" attacks to 
kill me (around 14 turns for the fox).
```

Original comment by `Grok4grok@gmail.com` on 19 Jan 2011 at 7:42

Attachments:
- [savegame1](https://storage.googleapis.com/google-code-attachments/andors-trail/issue-132/comment-4/savegame1)

**Comment by GoogleCodeExporter** [comment]
```
I just recreated the issue on your savegame, then reloaded my savegames after 
seeing that I the way I had tried to reproduce the problem was not quite 
accurate. I am now able to reproduce the problem on my savegames as well. So it 
is not limited to you.

It appears that interrupting the battle sequence is a bad idea at the moment.. 
don't do that and wait your turn. (lol)

I'm trying to find my way through the code to the actual battle sequence, from 
there I'll see if I can make some suggestions.
```

Original comment by `divilsp...@gmail.com` on 19 Jan 2011 at 9:03

**Comment by GoogleCodeExporter** [comment]
```
Thanks for the report. 

Yes, this is a bug. When starting a new Android Activity (such as the view item 
Activity), the game is actually saved and restored when you return to the game 
screen. This is similar to what happens if you for example leave to another 
program on your device, or take a phone call.

Savegames currently only support saving the state between turns.

I agreee that it is low prio to fix it, but a bug nonetheless. Thanks for 
reporting it.

Also, thanks for the update on "chitin". I'll see what I can do to update that.
```

Original comment by `oskar.wi...@gmail.com` on 29 Jan 2011 at 10:11

**Comment by GoogleCodeExporter** [comment]
```
Thank you both for your responses and help with this. I can see how the process 
that Oskar described would produce the results that I'm seeing.

I'm glad that I could contribute to the project in some (small) way.
```

Original comment by `Grok4grok@gmail.com` on 30 Jan 2011 at 5:01

**Comment by GoogleCodeExporter** [comment]
Original comment by `sdeva...@gmail.com` on 10 Feb 2011 at 2:08
- Changed state: **Accepted**
- Added labels: **Priority-Low**
- Removed labels: **Priority-Medium**

**Comment by GoogleCodeExporter** [comment]
```
As of v0.6.10, the game supports saving monster's current AP, which means that 
this is theoretically solvable now.

With the bugfixes done in r237, this should now be fully resolved.
```

Original comment by `oskar.wi...@gmail.com` on 29 Apr 2012 at 9:49
- Changed state: **Ready**
- Added labels: **Milestone-0.6.11**

**Comment by GoogleCodeExporter** [comment]
```
Released in v0.6.11. Closing issue.
```

Original comment by `oskar.wi...@gmail.com` on 9 Jun 2012 at 9:31
- Changed state: **Fixed**
